backend/pkg/api/handlers/input_validation.go

```diff
@@ -416,6 +435,17 @@ func checkSort[T enums.EnumFactory[T]](v *validationError, sortString string) *t
 	return &types.Sort[T]{Column: sortCol, Desc: desc}
 }
 
+func checkSearch[T enums.Searchable](searchString string) (T, error) {
```

Comment:
This func should prob never return an error, in worst case the returning searchable should be empty

Reply:
It's returning a bool now to indicate an `invalid search` (if `false`), aka the search query didn't match any accepted patterns for this search. This may also check extra conditions, e.g. if the `aggregateGroups` flag is set, the group search is not applied even if the regex would otherwise match. See the three current examples, e.g. https://github.com/gobitfly/beaconchain/pull/1014/files#diff-eeb9062bd8258c1d32e584699f9dc786ad81df60981ec50fbc17b3c8962d7581R666-R669

Motivation for this approach was to allow early-outing with empty responses on the api layer already if possible, before even invoking data access at all.